starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,93 @@
+
+const express = require("express");
+const authRoutes = express.Router();
+
+// User model
+const User           = require("../models/user");
+
+// BCrypt to encrypt passwords
+const bcrypt         = require("bcrypt");
+const bcryptSalt     = 10;
+
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+authRoutes.get("/login", (req, res, next) => {
+  res.render("auth/login");
+});
+
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+  const salt     = bcrypt.genSaltSync(bcryptSalt);
+  const hashPass = bcrypt.hashSync(password, salt);
+
+  if (username === "" || password === "") {
+    res.render("auth/signup", {
+      errorMessage: "Indicate a username and a password to sign up"
+    });
+    return;
+  }
+
+  User.findOne({ "username": username },
+  "username",
+  (err, user) => {
+    if (user !== null) {
+      res.render("auth/signup", {
+        errorMessage: "The username already exists"
+      });
+      return;
+    }
+
+    var salt     = bcrypt.genSaltSync(bcryptSalt);
+    var hashPass = bcrypt.hashSync(password, salt);
```

Comment:
You are writing these two lines twice (here and in lines 23, 24). Better put them here (no point in encrypt the password if you are not going to need it), but with const.